plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -196,53 +222,75 @@ internal class LocationPuckManager(
+      }
+    }
+  }
+
+  private fun get3DPuckScaleExpression(puck: LocationPuck3D, mercatorScale: Double = 1.0): Value {
+    val modelScaleConstant = 2.0.pow(MAX_ZOOM - MIN_ZOOM)
+    val modelScaleExpression = puck.modelScaleExpression
+    return if (modelScaleExpression == null) {
+      Value(
+        arrayListOf(
+          Value("interpolate"),
+          Value(arrayListOf(Value("exponential"), Value(PUCK_3D_EXPONENTIAL_EXPRESSION_BASE))),
+          Value(arrayListOf(Value("zoom"))),
+          Value(MIN_ZOOM),
           Value(
             arrayListOf(
-              Value("interpolate"),
-              Value(arrayListOf(Value("exponential"), Value(0.5))),
-              Value(arrayListOf(Value("zoom"))),
-              Value(MIN_ZOOM),
+              Value("literal"),
               Value(
                 arrayListOf(
-                  Value("literal"),
-                  Value(
-                    arrayListOf(
-                      Value(modelScaleConstant * puck.modelScale[0].toDouble()),
-                      Value(modelScaleConstant * puck.modelScale[1].toDouble()),
-                      Value(modelScaleConstant * puck.modelScale[2].toDouble())
-                    )
-                  )
+                  Value(modelScaleConstant * puck.modelScale[0].toDouble() * mercatorScale),
+                  Value(modelScaleConstant * puck.modelScale[1].toDouble() * mercatorScale),
+                  Value(modelScaleConstant * puck.modelScale[2].toDouble() * mercatorScale)
                 )
-              ),
-              Value(MAX_ZOOM),
+              )
+            )
+          ),
+          Value(MAX_ZOOM),
+          Value(
+            arrayListOf(
+              Value("literal"),
               Value(
                 arrayListOf(
-                  Value("literal"),
-                  Value(
-                    arrayListOf(
-                      Value(puck.modelScale[0].toDouble()),
-                      Value(puck.modelScale[1].toDouble()),
-                      Value(puck.modelScale[2].toDouble())
-                    )
-                  )
+                  Value(puck.modelScale[0].toDouble() * mercatorScale),
+                  Value(puck.modelScale[1].toDouble() * mercatorScale),
+                  Value(puck.modelScale[2].toDouble() * mercatorScale)
                 )
               )
             )
           )
-        } else {
-          Value.fromJson(modelScaleExpression).take()
-        }
-        locationLayerRenderer.styleScaling(scaleExpression)
-      }
+        )
+      )
+    } else {
+      Value.fromJson(modelScaleExpression).take()
     }
   }
 
+  private fun mercatorScale(lat: Double): Double {
+    // In Mercator projection the scale factor is changed along the meridians as a function of latitude
+    // to keep the scale factor equal in all direction: k=sec(latitude), where sec(α) = 1 / cos(α).
+    return cos(
```

Comment:
docs state from what I see `1 / cos(a)` but we do actually calculate simply `cos(a)` - is this intended?

Reply:
yes, we are trying to inverse the logic, as the puck is already real-world size.

Reply:
Updated the doc as well.